test/e2e/azure_accelerated_networking.go

```diff
@@ -0,0 +1,86 @@
+// +build e2e
+
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"path/filepath"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-12-01/compute"
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2019-06-01/network"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+type AzureAcceleratedNetworkingSpecInput struct {
+	ClusterName string
+}
+
+func AzureAcceleratedNetworkingSpec(ctx context.Context, inputGetter func() AzureAcceleratedNetworkingSpecInput) {
+	var (
+		specName = "azure-accelerated-networking"
+		input    AzureAcceleratedNetworkingSpecInput
+	)
+
+	input = inputGetter()
+	Expect(input.ClusterName).NotTo(BeEmpty(), "Invalid argument. input.ClusterName can't be empty when calling %s spec", specName)
+
+	By("creating Azure clients with the workload cluster's subscription")
+	settings, err := auth.GetSettingsFromEnvironment()
+	Expect(err).NotTo(HaveOccurred())
+	subscriptionID := settings.GetSubscriptionID()
+	authorizer, err := settings.GetAuthorizer()
+	Expect(err).NotTo(HaveOccurred())
+	vmsClient := compute.NewVirtualMachinesClient(subscriptionID)
+	vmsClient.Authorizer = authorizer
+	nicsClient := network.NewInterfacesClient(subscriptionID)
+	nicsClient.Authorizer = authorizer
+
+	By("verifying EnableAcceleratedNetworking for the primary NIC of each VM")
+	acceleratedNetworking := map[compute.VirtualMachineSizeTypes]bool{
```

Comment:
what is this list? A subset of SKUs that we expect to run tests with? Does that mean this test would fail if it was run with any other vmSize?

Reply:
Yes, exactly. Rather than (re-)implement the Azure API calls to find the accelerated networking capability, I added a static lookup table. I can populate it with all known SKUs if we like, these are just the ones that I can see referenced in CAPZ code.